java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/vector/VectorCollectionPutDatasetTest.java

```diff
@@ -1,155 +1,97 @@
-                        .toCompletableFuture()
-                        .join()
-        );
+        var vector = reader.getTrainVector(index % testDataSetSize);
+        collection.putAsync(index, VectorDocument.of(index % testDataSetSize, VectorValues.of(vector)))
+                .toCompletableFuture()
+                .join();
     }
 
-    @TimeStep(prob = 1)
-    public void putAll(ThreadState state) {
+    @TimeStep(prob = 0)
+    public void putAll() {
+        int testDataSetSize = reader.getSize();
+        var size = getRequestedSize();
+
         var iteration = counter.getAndAdd(putBatchSize);
-        if (iteration >= loadFirst) {
+        if (iteration >= size) {
             testContext.stop();
             return;
         }
         Map<Integer, VectorDocument<Integer>> buffer = new HashMap<>();
-        metrics.recordBuffer(
-                () -> {
-                    for (int i = 0; i < putBatchSize; i++) {
-                        var key = iteration + i;
-                        if (key >= reader.size) {
-                            break;
-                        }
-                        var vector = reader.getTrainVector(key);
-                        buffer.put(key, VectorDocument.of(key, VectorValues.of(vector)));
-                    }
-                }
-        );
-
-        metrics.recordPut(
-                () -> collection.putAllAsync(buffer)
-                        .toCompletableFuture()
-                        .join()
-        );
-    }
-
-    @AfterRun
-    public void afterRun() {
-        logger.info("****CUSTOM STATISTICS****");
-        logger.info(metrics.getStatistics());
-    }
+        for (int i = 0; i < putBatchSize; i++) {
+            var key = iteration + i;
+            if (key >= testDataSetSize) {
+                break;
+            }
+            var vector = reader.getTrainVector(key % testDataSetSize);
+            buffer.put(key, VectorDocument.of(key, VectorValues.of(vector)));
+        }
 
-    public static class ThreadState extends BaseThreadState {
+        collection.putAllAsync(buffer)
+                .toCompletableFuture()
+                .join();
     }
 
+    /**
+     * Deletes entries 0..requested size
+     */
+    @TimeStep(prob = 0)
+    public void delete() {
+        var size = getRequestedSize();
 
-    public static class TimeMetrics {
-        private static final Histogram bufferTimer = new Histogram(2);
-        private static final Histogram putTimer = new Histogram(2);
-
-
-        public void recordBuffer(Runnable action) {
-            var start = System.currentTimeMillis();
-            action.run();
-            bufferTimer.recordValue(System.currentTimeMillis() - start);
-        }
-
-        public void recordPut(Runnable action) {
-            var start = System.currentTimeMillis();
-            action.run();
-            putTimer.recordValue(System.currentTimeMillis() - start);
+        var index = counter.getAndIncrement();
```

Comment:
I’m not sure about the purpose of the optimize and delete tests. Running them individually (prob=1) doesn’t make sense since the database would be empty, with nothing to delete or optimize. So, they wouldn’t measure the performance of delete and optimize operation.

Running these as part of several simultaneous operations (prob < 1) is still unclear since ‘var index = counter.getAndIncrement();’ might target an object that hasn’t been added yet. It would only make sense if the size exceeds the training dataset size, but even then, it would affect the performance results of the delete operation.

Reply:
I use scenarios like (note that this is single test case `pouo`):

```
# put_optimize_update_optimize
- name: pouo
  <<: *defaults
  test:
    - class: com.hazelcast.simulator.tests.vector.VectorCollectionPutDatasetTest
      name: t01_loadInitialData
      <<: *collection_params
      threadCount: 100
      logRateMs: 60_000
      putProb: 1
#      putAllProb: 1
    - class: com.hazelcast.simulator.tests.vector.VectorCollectionPutDatasetTest
      name: t02_optimizeAfterLoad
      <<: *collection_params
      threadCount: 1
      logRateMs: 60_000
      optimizeProb: 1
    - class: com.hazelcast.simulator.tests.vector.VectorCollectionPutDatasetTest
      # This requires no deduplication to take and effect
      name: t03_updateTenPercent
      <<: *collection_params
      threadCount: 100
      logRateMs: 60_000
      putProb: 1
      targetCollectionSizeFraction: 0.1
    - class: com.hazelcast.simulator.tests.vector.VectorCollectionPutDatasetTest
      name: t04_optimizeAfterUpdates
      <<: *collection_params
      threadCount: 1
      logRateMs: 60_000
      optimizeProb: 1
```

All steps execute sequentially on the same cluster.

I will push shortly the scenarios to the benchmarks repo.